hbase-client/src/main/java/org/apache/hadoop/hbase/client/Admin.java

```diff
@@ -1064,7 +1067,31 @@ default ServerName getMaster() throws IOException {
    * @throws IOException if a remote or network exception occurs
    */
   default Collection<ServerName> getRegionServers() throws IOException {
-    return getClusterMetrics(EnumSet.of(Option.SERVERS_NAME)).getServersName();
+    return getRegionServers(false);
+  }
+
+  /**
+   * Retrieve all current live region servers including decommissioned
+   * if excludeDecommissionedRS is false, else non-decommissioned ones only
+   *
+   * @param excludeDecommissionedRS should we exclude decommissioned RS nodes
+   * @return all current live region servers including/excluding decommissioned hosts
+   * @throws IOException if a remote or network exception occurs
+   */
+  default Collection<ServerName> getRegionServers(boolean excludeDecommissionedRS)
+      throws IOException {
+    List<ServerName> allServers =
+      getClusterMetrics(EnumSet.of(Option.SERVERS_NAME)).getServersName();
+    if (!excludeDecommissionedRS) {
+      return allServers;
+    }
+    List<ServerName> decommissionedRegionServers = listDecommissionedRegionServers();
+    if (CollectionUtils.isNotEmpty(decommissionedRegionServers)) {
```

Comment:
```
 if (CollectionUtils.isNotEmpty(decommissionedRegionServers)) {
    allServers = new ArrayList<>(allServers);
    allServers.removeIf(decommissionedRegionServers::contains);
    return Collections.unmodifiableList(allServers);
  }
 return allServers;
```
to
```
  return allServers.stream()
    .filter(s -> !decommissionedRegionServers.contains(s))
    .collect(ImmutableList.toImmutableList());
```

I think this will save one time full copy (previous code copy twice, and this copies just once). Need to confirm.

Reply:
The reason why I am copying (`new ArrayList<>(allServers)`) is because `allServers` is unmodifiable list